packages/sw-background-sync-queue/src/lib/queue-utils.js

```diff
@@ -0,0 +1,94 @@
+import {getDb} from './background-sync-idb-helper';
+
+/**
+ * takes a request and gives back JSON object that is storable in IDB
+ *
+ * @param {Request} request request object to transform
+ * into iDB storable object
+ * @param {Object} config config object to be
+ * stored along in the iDB
+ * @return {Object} indexable object for iDB
+ *
+ * @memberOf RequestManager
+ */
+async function getQueueableRequest(request, config) {
+	let requestObject={
+		config: config,
+		metadata: {
+			creationTimestamp: Date.now(),
+		},
+	};
+	let headerObject = {};
```

Comment:
Oh, I was looking into this more, and you can serialize the headers in a straightforward manner using the approach at http://www.2ality.com/2015/08/es6-map-json.html

```
const serializedHeaders = JSON.stringify([...r.headers]);
```

And then you can deserialize using

```
const deserializedHeaders = new Headers(JSON.parse(serializedHeaders));
```

since the `Headers` constructor [knows](https://fetch.spec.whatwg.org/#example-headers-class) how to deal with arrays of ordered pairs.